packages/components/badge/src/BadgeItem.tsx

```diff
@@ -0,0 +1,79 @@
+import { forwardRef, HTMLAttributes, PropsWithRef } from 'react'
+
+import { styles, type StylesProps } from './BadgeItem.styles'
+import {
+  DEFAULT_INTENT,
+  DEFAULT_LABELS,
+  DEFAULT_OVERFLOW_COUNT,
+  DEFAULT_SIZE,
+  DEFAULT_TYPE,
+} from './config'
+
+interface ACCESSIBILITY_LABELS {
+  noCount?: string
+  singleCount?: string
+  pluralCount?: string
+  moreThanOverflowCount?: string
+}
+
+export interface BadgeItemProps extends PropsWithRef<HTMLAttributes<HTMLSpanElement>>, StylesProps {
+  /**
+   * Numeric value used as indicator inside the component.
+   */
+  count?: number
+  /**
+   * Maximum numeric value to be dispayed as a count value.
+   */
+  overflowCount?: number
+  /**
+   * Custom labels for accessibility purposes than can be set for each count case.
+   */
+  labels?: ACCESSIBILITY_LABELS
```

Comment:
Not sure it's the best approach for managing labels for a11y... any suggestions?

Reply:
Maybe a `label` prop. The prop should be a function that takes a `string` (parsed count) a returns a `string` that will be used as `aria-label`. This way users can use they i18n library to do the pluralization magic

Reply:
Perhaps role status is needed (see https://www.w3.org/WAI/WCAG22/Techniques/aria/ARIA22 and https://stackoverflow.com/questions/68170589/do-html-elements-like-pill-badge-need-extra-aria-role)

Reply:
> Maybe a `label` prop. The prop should be a function that takes a `string` (parsed count) a returns a `string` that will be used as `aria-label`. This way users can use their favourite i18n library to do the formatting

Yep, I was thinking on that solution, but that way users will need to build on their own the logic behind the design specs we have in ZH right now, not perfect solution I guess...

Reply:
Role added ;)

Reply:
But this would only apply for the `label` and the function is going to provide the parsed count (for example, `+99` and perhaps) so user can do something like this:
```
<Badge label={count => i18n.t("key", {count})} />
```

Reply:
Type: string | ({ [INTERNAL_DATA]: string }) => string

<Badge aria-label={({ count, overflowCount }) => i18n.t("key", {count})} />